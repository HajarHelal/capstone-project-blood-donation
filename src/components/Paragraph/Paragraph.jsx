import React from 'react';
import { PropTypes } from 'prop-types';

function Paragraph({ headerText, paragraphText, headerCenter, ParagraphCenter }) {
  const classesObjectStyle = {
    containerStyle: 'bg-gray-200 w-screen',
    textDivStyle: ' w-11/12 ml-12 ',
    headisCentered: headerCenter ? 'text-center' : '',
    textStyle: 'text-primary-100 font-bold pt-5 pb-5',
    ParagraphisCentered: ParagraphCenter ? 'text-center' : '',
  };
  return (
    <div className={classesObjectStyle.containerStyle}>
      <div className={classesObjectStyle.textDivStyle}>
        <h1 className={classesObjectStyle.textStyle}>{headerText}</h1>
        <p className={classesObjectStyle.ParagraphisCentered}>{paragraphText}</p>
      </div>
    </div>
  );
}

Paragraph.propTypes = {
  headerText: PropTypes.string.isRequired,
  paragraphText: PropTypes.string.isRequired,
  headerCenter: PropTypes.bool,
  ParagraphCenter: PropTypes.bool,
};
export default Paragraph;
