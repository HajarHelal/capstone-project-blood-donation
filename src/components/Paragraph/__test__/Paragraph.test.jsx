import React from 'react';
import renderer from 'react-test-renderer';
import Paragraph from '../Paragraph';

it('renders correctly', () => {
  const tree = renderer
    .create(<Paragraph headerText paragraphText headerCenter ParagraphCenter />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
