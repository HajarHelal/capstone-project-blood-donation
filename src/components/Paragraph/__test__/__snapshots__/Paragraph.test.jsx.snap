// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1
    className="text-primary-100  sm:text-xl md:text-2xl lg:text-3xl pt-1 font-semibold  text-center"
  />
  <p
    className=" pt-3 pb-5  sm:text-sm md:text-base lg:text-lg  text-center "
  />
</div>
`;
